main/src/test/java/org/mobilitydata/gtfsvalidator/util/CalendarUtilTest.java

```diff
@@ -0,0 +1,204 @@
+                .containsExactly(
+                        "s1",
+                        new ServicePeriod(LocalDate.of(2021, 1, 4),
+                                LocalDate.of(2021, 1, 5), (byte) 0b0010111,
+                                ImmutableSet.of(LocalDate.of(2021, 1, 6)),
+                                ImmutableSet.of(LocalDate.of(2021, 1, 7))),
+                        "s2",
+                        new ServicePeriod(LocalDate.of(2021, 2, 14),
+                                LocalDate.of(2021, 2, 15), (byte) 0b1000000,
+                                ImmutableSet.of(), ImmutableSet.of()),
+                        "s3",
+                        new ServicePeriod(LocalDate.EPOCH, LocalDate.EPOCH, (byte) 0,
+                                ImmutableSet.of(LocalDate.of(2021, 3, 8)),
+                                ImmutableSet.of(LocalDate.of(2021, 3, 9))));
+    }
+
+    @Test
+    public void servicePeriodToServiceDatesMap() {
+        assertThat(
+                CalendarUtil.servicePeriodToServiceDatesMap(ImmutableMap.of(
+                        "s1",
+                        new ServicePeriod(LocalDate.of(2021, 1, 4),
+                                LocalDate.of(2021, 1, 5), (byte) 0b0010111,
+                                ImmutableSet.of(LocalDate.of(2021, 1, 6)),
+                                ImmutableSet.of(LocalDate.of(2021, 1, 7))),
+                        "s2",
+                        new ServicePeriod(LocalDate.of(2021, 2, 14),
+                                LocalDate.of(2021, 2, 15), (byte) 0b1000000,
+                                ImmutableSet.of(), ImmutableSet.of()),
+                        "s3",
+                        new ServicePeriod(LocalDate.EPOCH, LocalDate.EPOCH, (byte) 0,
+                                ImmutableSet.of(LocalDate.of(2021, 3, 8)),
+                                ImmutableSet.of(LocalDate.of(2021, 3, 9))))))
+                .containsExactly("s1",
+                        ImmutableSortedSet.of(LocalDate.of(2021, 1, 4),
+                                LocalDate.of(2021, 1, 5),
+                                LocalDate.of(2021, 1, 6)),
+                        "s2", ImmutableSortedSet.of(LocalDate.of(2021, 2, 14)),
+                        "s3", ImmutableSortedSet.of(LocalDate.of(2021, 3, 8)));
+    }
+
+    @Test
+    public void firstIntersectingDate() {
+        assertThat(CalendarUtil.firstIntersectingDate(
+                ImmutableSortedSet.of(LocalDate.of(2021, 1, 4),
+                        LocalDate.of(2021, 1, 5),
+                        LocalDate.of(2021, 1, 6)),
+                ImmutableSortedSet.of(
+                        LocalDate.of(2021, 1, 1), LocalDate.of(2021, 1, 5),
+                        LocalDate.of(2021, 1, 6), LocalDate.of(2021, 1, 7))))
+                .isEqualTo(Optional.of(LocalDate.of(2021, 1, 5)));
+
+        assertThat(CalendarUtil.firstIntersectingDate(
```

Comment:
To ensure the potentially endless `for` loop always terminates I believe you should also test the case where the first set contains the largest `LocalDate` value. So this, but inverting the arguments into `firstIntersectingDate()`

Reply:
Done.